vmdb/spec/models/vmdb_database_connection_spec.rb

```diff
@@ -0,0 +1,75 @@
+require "spec_helper"
+
+describe VmdbDatabaseConnection do
+  before :each do
+    @db = FactoryGirl.create(:vmdb_database)
+  end
+
+  it 'can find connections' do
+    connections = VmdbDatabaseConnection.all
+    expect(connections.length).to be > 0
+  end
+
+  it 'is blocked' do
+    locked_latch = ActiveSupport::Concurrency::Latch.new
+    continue_latch = ActiveSupport::Concurrency::Latch.new
+
+    get_lock = Thread.new do
+      VmdbDatabaseConnection.connection.transaction do
+        VmdbDatabaseConnection.connection.execute('LOCK users IN EXCLUSIVE MODE')
+        locked_latch.release
+        continue_latch.await
+      end
+    end
+
+    wait_for_lock = Thread.new do
+      VmdbDatabaseConnection.connection.transaction do
+        locked_latch.await # wait until `get_lock` has the lock
+        VmdbDatabaseConnection.connection.execute('LOCK users IN EXCLUSIVE MODE')
+      end
+    end
+
+    locked_latch.await # wait until `get_lock` has the lock
+    # spin until `wait_for_lock` is waiting to acquire the lock
+    loop { break if wait_for_lock.status == "sleep" }
+
+    connections = VmdbDatabaseConnection.all
+
+    blocked_conn = connections.find(&:blocked_by)
```

Comment:
We generally prefer `.detect` over `.find`.  Personally I get so confused when .find is used because this looks like you're doing an ActiveRecord#find on that relation.